src/App.tsx

```diff
@@ -1,61 +1,112 @@
 import './App.scss';
+import { FC, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import usersFromServer from './api/users';
+import { getUser } from './helpers/getUser';
+import { todosList } from './helpers/todosList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const App: FC = () => {
+  const [todos, setTodos] = useState([...todosList]);
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [title, setTitle] = useState('');
+  const [titleErrorField, seTtitleErrorField] = useState(false);
+  const [selectErrorField, setSelectErrorField] = useState(false);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      seTtitleErrorField(true);
+    }
+
+    if (!selectedUser) {
+      setSelectErrorField(true);
+    }
+
+    if (title && selectedUser) {
+      const createId = (arr:Todo[]) => {
+        const ids = arr.map(item => item.id);
+
+        return Math.max(...ids) + 1;
+      };
+
+      setTodos(prevTodos => {
+        const newTodo: Todo = {
+          id: createId(prevTodos),
+          title,
+          completed: false,
+          userId: selectedUser,
+          user: getUser(selectedUser),
+        };
+
+        return [...prevTodos, newTodo];
+      });
+
+      setTitle('');
+      setSelectedUser(0);
+    }
```

Comment:
- if we invert the condition, we will improve  code readability.
- I think you can create a separate helper function for `createId`


```suggestion
    if (!title && !selectedUser) {
      return;
    }
      const createId = (arr:Todo[]) => {
        const ids = arr.map(item => item.id);

        return Math.max(...ids) + 1;
      };

      setTodos(prevTodos => {
        const newTodo: Todo = {
          id: createId(prevTodos),
          title,
          completed: false,
          userId: selectedUser,
          user: getUser(selectedUser),
        };

        return [...prevTodos, newTodo];
      });

      setTitle('');
      setSelectedUser(0);
```